src/Utilities/OperatingSystemFilterHelpers.test.js

```diff
@@ -118,82 +64,82 @@ describe('groupOSFilterVersions', () => {
   });
 
-  it('returns group configuration with 3 groups and 4 values', () => {
-    const config = groupOSFilterVersions(testVersions);
+  it('returns group configuration with 5 groups', () => {
+    const config = groupOSFilterVersions(availableVersions);
     expect(config).toMatchSnapshot();
     expect(config[0].groupSelectable).toEqual(true);
     expect(config[0].label).toEqual('RHEL 8');
-    expect(config[0].value).toEqual('8');
+    expect(config[0].value).toEqual('RHEL 8');
   });
 });
 
 describe('buildOSFilterChip', () => {
-  it('returns two version chips', () => {
-    const chips = buildOSFilterChip(testValue, testVersions);
+  it('returns five version chips', () => {
+    const chips = buildOSFilterChip(testValue2, availableVersions);
     expect(chips).toMatchSnapshot();
-    expect(chips[0].chips).toHaveLength(3);
+    expect(chips[0].chips).toHaveLength(5);
   });
 });
 
 describe('onOSFilterChange', () => {
   const clickedGroup = {
-    value: '8',
+    value: 'RHEL 8',
   };
   const clickedItem = {
-    value: '8.5',
+    value: '8.10',
   };
 
   it('returns new selection with group selection enabled', () => {
-    const selection = cloneDeep(testValue);
-    set(selection, [8, 8], false);
+    const selection = cloneDeep(testValue3);
+    selection['RHEL 8'] = {
+      // eslint-disable-next-line prettier/prettier
+      '8.5': true,
+      '8.10': true,
+    };
     expect(
       onOSFilterChange(undefined, selection, clickedGroup, clickedItem)
-    ).toEqual(testValue);
+    ).toEqual(testValue4);
   });
 
   it('returns new selection with group selection disabled', () => {
-    const selection = cloneDeep(testValue);
-    selection[8][8.1] = false;
-    const expected = cloneDeep(selection);
-    expected[8][8] = false;
+    const selection = cloneDeep(testValue4);
+    selection['RHEL 8'] = {
+      // eslint-disable-next-line prettier/prettier
+      '8.5': false,
+      '8.10': false,
+    };
     expect(
       onOSFilterChange(undefined, selection, clickedGroup, clickedItem)
-    ).toEqual(expected);
+    ).toEqual(testValue3);
   });
 
   it('returns new selection with all minors selected after click on group', () => {
     const clickedItem = {
-      value: '8',
+      value: 'RHEL 8',
     };
-    const selection = cloneDeep(testValue);
-    selection[8] = {
-      8: true,
+    const selection = cloneDeep(testValue3);
+    selection['RHEL 8'] = {
+      'RHEL 8': true,
       8.5: false,
       '8.10': false,
     };
     expect(
       onOSFilterChange(undefined, selection, clickedGroup, clickedItem)
-    ).toEqual(testValue);
+    ).toEqual(testValue4);
   });
 
   it('returns new selection with all minors disabled after click on group', () => {
     const clickedItem = {
-      value: '8',
+      value: 'RHEL 8',
     };
-    const selection = cloneDeep(testValue);
-    selection[8] = {
-      8: false,
+    const selection = cloneDeep(testValue4);
+    selection['RHEL 8'] = {
+      'RHEL 8': false,
```

Comment:
Shouldn't all key here get an OS name prefixed?

Reply:
Not sure what you mean here. `'RHEL 8': false` is the key value pair needed to check/uncheck the entire group.

Reply:
I mean for consistency, shouldn't all the keys on the following lines also get a OS name prefix? Otherwise it might be confusing later for someone, why some do and some don't have an OS name attached.

Reply:
After understanding it now, it is still odd to me. The separate state for the major version itself seems redundant, since we can assume that if all minor versions are selected the group will be selected, otherwise it'll always be false, or?